docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,734 @@
+     -
+   * - :class:`~qiskit.opflow.PauliOp`
+     - :class:`~qiskit.quantum_info.Pauli`
+     - For direct compatibility with classes in :mod:`~qiskit.algorithms`, wrap in :class:`~qiskit.quantum_info.SparsePauliOp`
+   * - :class:`~qiskit.opflow.PauliSumOp`
+     - :class:`~qiskit.quantum_info.SparsePauliOp`
+     - See example below
+   * - :class:`~qiskit.opflow.TaperedPauliSumOp`
+     - This class was used to combine a :class:`~PauliSumOp` with its identified symmetries in one object. It has been deprecated without replacement
+     - See ``Z2Symmetries`` example for updated workflow
+   * - :class:`~qiskit.opflow.Z2Symmetries`
+     - :class:`~qiskit.quantum_info.Z2Symmetries`
+     - See example below
+
+
+PrimitiveOps Examples
+~~~~~~~~~~~~~~~~~~~~~
+.. list-table::
+   :header-rows: 1
+
+   * - opflow
+     - alternative
+     - notes
+
+   * -  ``PauliSumOp`` **Example:**
+
+        .. code-block:: python
+
+            from qiskit.opflow import PuliSumOp
+            from qiskit.quantum_info import SparsePauliOp, Pauli
+
+            qubit_op = PauliSumOp(SparsePauliOp(Pauli("XYZY"), coeffs=[2]), coeff=-3j)
+
+     -
+
+        .. code-block:: python
+
+            from qiskit.quantum_info import SparsePauliOp, Pauli
+
+            qubit_op = SparsePauliOp(Pauli("XYZY")), coeff=-6j)
+
+     -
+   * -  ``Z2Symmetries`` **and** ``TaperedPauliSumOp`` **Example:**
+
+        .. code-block:: python
+
+            from qiskit.opflow import PuliSumOp, Z2Symmetries, TaperedPauliSumOp
+
+            qubit_op = PauliSumOp.from_list(
+                [
+                ("II", -1.0537076071291125),
+                ("IZ", 0.393983679438514),
+                ("ZI", -0.39398367943851387),
+                ("ZZ", -0.01123658523318205),
+                ("XX", 0.1812888082114961),
+                ]
+            )
+            z2_symmetries = Z2Symmetries.find_Z2_symmetries(qubit_op)
+            tapered_op = z2_symmetries.taper(qubit_op)
+            # can be represented as:
+            tapered_op = TaperedPauliSumOp(primitive, z2_symmetries)
+     -
+
+        .. code-block:: python
+
+            from qiskit.quantum_info import SparsePauliOp, Z2Symmetries
+
+            qubit_op = SparsePauliOp.from_list(
+                [
+                    ("II", -1.0537076071291125),
+                    ("IZ", 0.393983679438514),
+                    ("ZI", -0.39398367943851387),
+                    ("ZZ", -0.01123658523318205),
+                    ("XX", 0.1812888082114961),
+                ]
+            )
+            z2_symmetries = Z2Symmetries.find_z2_symmetries(qubit_op)
+            tapered_op = z2_symmetries.taper(qubit_op)
+     -
+
+
+ListOps
+~~~~~~~
+.. list-table:: Migration of ``qiskit.opflow.list_ops``
+   :header-rows: 1
+
+   * - opflow
+     - alternative
+     - notes
+
+   * - :class:`~qiskit.opflow.ListOp`
+     - No direct replacement. This class was used internally within opflow.
```

Comment:
I would delete this bit for many cases here - `This class was used internally within opflow.` leaving just the fact that there is no direct replacement. These classes are exposed public classes for end users. Sometimes they did get built implicitly through the opflow expressions, but you could do it explicilitly - this https://qiskit.org/documentation/tutorials/operators/01_operator_flow.html for instance builds out ListOps and with expressions a ComposedOp etc.

Now we can compose and tensor operators. opflow was designed for lazy evaluation so its a little different but its similar. The SummedOp, if you have a sum of Paulis, which is what it was used for to start with mostly, before we did the more efficient PauliSumOp (based on SparsePauliOp) - then a reasonable alternative for those common cases is the SparsePauliOp. More generally the SummedOp could contain any other opflow operator so it general there is no alternative.